plugins/woocommerce/includes/class-wc-session-handler.php

```diff
@@ -697,6 +693,45 @@ public function update_session_timestamp( $customer_id, $timestamp ) {
 		$GLOBALS['wpdb']->update( $this->_table, array( 'session_expiry' => $timestamp ), array( 'session_key' => $customer_id ), array( '%d' ) );
 	}
 
+	/**
+	 * Destroys the WooCommerce session if it contains no data for non-logged-in users.
+	 *
+	 * This method helps improve caching performance by removing session cookies when they
+	 * are no longer needed, allowing non-logged-in customers to receive cached pages.
+	 * Only runs if the destroy-empty-sessions feature is enabled.
+	 *
+	 * @return void
+	 *
+	 * @since 10.3.0
+	 */
+	public function destroy_session_if_empty() {
```

Comment:
Given that this method exists exclusively to be the target of a hook (for now at least), could we mark it with an `@internal` tag (or more explicitly, `@internal For exclusive usage of WooCommerce core, backwards compatibility not guaranteed` as we have done in other places)? We can always remove the tag in the future if we see value in exposing it to plugins.